### ISSUE TITLE
Retry logic for running pub [get, etc]

### ISSUE THREAD

**Issue by sethladd** [financeCoding/dart-carte-du-jour#90]
Looking at the logs, I see SSL errors as a reason for build failures. Check out:

`Handshake error in client (OS Error: SSL peer reports incorrect Message Authentication Code., errno = -12272)`

in http://www.dartdocs.org/buildlogs/b-ae6665f19488c940fa2c1f1f024e5-startupscript.log

and `SSL received a record with an incorrect Message Authentication Code. error -12273` in http://www.dartdocs.org/buildlogs/b-5f914e95e441991c8315e4992b9d8-startupscript.log

I wonder if we can retry the `pub cache` or `pub get` logic in `buildDocumentationCacheSync` ?

Basically, how do we handle transient issues?
